### PR TITLE
Task s1 handle login response

### PR DESCRIPTION
**Pull request by karimkefi**

### CODE REVIEW

File: public/js/loginValidation.js [hunk 1]
```diff
@@ -17,30 +17,41 @@ let validateInputs = () => {
...
+    return response
+
 }
 
 let loginForm = document.getElementById('loginForm')
-loginForm.addEventListener('submit', (e) => {
-    let validInputs
+loginForm.addEventListener('submit', async (e) => {
     e.preventDefault()
+    let validInputs
```
Comment (nsantiagoblair) [inline public/js/loginValidation.js]: why are you assigning here and declaring on the following line?

File: public/js/loginValidation.js [hunk 2]
```diff
@@ -17,30 +17,41 @@ let validateInputs = () => {
...
+            credentials: "same-origin",
             headers: {
                 'Accept': 'application/json',
-                'Content-Type': 'application/json'
+                'Content-Type': 'application/json',
             },
             method: "POST",
             body: JSON.stringify(data)
         })
         .then(function(data){return data.json()})
```
Comment (nsantiagoblair) [inline public/js/loginValidation.js]: ES6? 😿 
`.then(data => data.json())` beeeutiful!! 😸

File: public/js/loginValidation.js [hunk 3]
```diff
@@ -17,30 +17,41 @@ let validateInputs = () => {
...
+        .then(function(data){return data})
+    return response
```
Comment (nsantiagoblair) [inline public/js/loginValidation.js]: could just return the fetch.. `return fetch()`
Only using the variable to return it

File: public/js/loginValidation.js [hunk 4]
```diff
@@ -17,30 +17,41 @@ let validateInputs = () => {
     let cleanedPasswordInput = encodeURI(document.getElementById('password').value)
 
     if (isValidEmail('userEmail')) {
-        let emailPasswordValues = {'email' : cleanedEmailInput, 'password': cleanedPasswordInput}
+        let emailPasswordValues = {'userEmail' : cleanedEmailInput, 'password': cleanedPasswordInput}
         return emailPasswordValues
```
Comment (nsantiagoblair) [inline public/js/loginValidation.js]: Could just return the object. You're only using emailPasswordValues variable to return it...

File: public/js/loginValidation.js [hunk 5]
```diff
@@ -17,30 +17,41 @@ let validateInputs = () => {
...
+loginForm.addEventListener('submit', async (e) => {
     e.preventDefault()
+    let validInputs
     validInputs = validateInputs()
-    sendLoginDetails(validInputs)
-})
+    let response = await sendLoginDetails('login', validInputs)
+
+    if(!response['success']) {
+        document.getElementById("error-message").innerText = response['msg']
```
Comment (nsantiagoblair) [inline public/js/loginValidation.js]: I don't think the user should be seeing the error message the fetch produces...

File: public/js/loginValidation.js [hunk 6]
```diff
@@ -17,30 +17,41 @@ let validateInputs = () => {
...
+    } else if (response['success'] === true) {
+        window.location.href = "/admin";
+    }
```
Comment (nsantiagoblair) [inline public/js/loginValidation.js]: `let errorMsgBox = document.getElementById("error-message")`
`response['success'] === true ?`
 `window.location.href = "/admin" : errorMsgBox.innerText = response['msg']`
    🦅

File: templates/index.phtml [hunk 7]
```diff
@@ -24,5 +24,6 @@
                 <span id="submitWarning" class="req"></span>
             </div>
         </form>
+        <div id="error-message"></div>
```
Comment (nsantiagoblair) [inline templates/index.phtml]: if it pertains to the form i'd put this div inside the form

File: public/js/loginValidation.js (hunk 1 above)
Comment (karimkefi) [reply public/js/loginValidation.js]: yup

File: public/js/loginValidation.js (hunk 2 above)
Comment (karimkefi) [reply public/js/loginValidation.js]: ES5 ES5 ES5!!!

File: public/js/loginValidation.js (hunk 3 above)
Comment (karimkefi) [reply public/js/loginValidation.js]: we need to await validate inputs, before awiting the fetch, before awaiting sendloginDetails, so cannot immediately return fetch()

File: public/js/loginValidation.js (hunk 4 above)
Comment (karimkefi) [reply public/js/loginValidation.js]: fair point

File: public/js/loginValidation.js (hunk 6 above)
Comment (karimkefi) [reply public/js/loginValidation.js]: sweet

File: templates/index.phtml (hunk 7 above)
Comment (karimkefi) [reply templates/index.phtml]: done

**Pull request merged**